### PR TITLE
Add support for non-alphanumeric characters in type names

### PR DESCRIPTION
**Pull request by whitequark**

### CODE REVIEW

**Comment by seibert** [comment]
Seems reasonable to me.  @sklam: Any objections?

**Comment by sklam** [comment]
I think we don't need to replace space once we have the doublequote

**Comment by whitequark** [comment]
Hm right, I copied this from Value. It shouldn't be needed there either, let me update.

**Comment by whitequark** [comment]
Fixed.

File: llvmlite/ir/types.py [hunk 1]
```diff
@@ -17,6 +17,10 @@
 TYPE_METADATA = _type_enum()
 
 
+def _wrapname(x):
+    return '"{0}"'.format(x.replace('\\', '\\5c').replace('"', '\\22'))
```
Comment (sklam) [inline llvmlite/ir/types.py]: can you add tests for '\\' and '"' ?

Comment (whitequark) [inline llvmlite/ir/types.py]: Done

**Comment by whitequark** [comment]
Ping

**Comment by sklam** [comment]
Thanks!

**Pull request merged**